src/GovUk.Education.ExploreEducationStatistics.Content.Services/DataGuidanceService.cs

```diff
@@ -1,36 +1,39 @@
-﻿using System.Threading.Tasks;
+﻿#nullable enable
+using System.Threading.Tasks;
 using GovUk.Education.ExploreEducationStatistics.Common.Model;
 using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
 using GovUk.Education.ExploreEducationStatistics.Content.Services.ViewModels;
 using GovUk.Education.ExploreEducationStatistics.Data.Services.Interfaces;
-using GovUk.Education.ExploreEducationStatistics.Publisher.Model.ViewModels;
 using Microsoft.AspNetCore.Mvc;
 
 namespace GovUk.Education.ExploreEducationStatistics.Content.Services
 {
     public class DataGuidanceService : IDataGuidanceService
     {
-        private readonly IFileStorageService _fileStorageService;
         private readonly IDataGuidanceSubjectService _dataGuidanceSubjectService;
+        private readonly Interfaces.IPublicationService _publicationService;
+        private readonly Interfaces.IReleaseService _releaseService;
 
         public DataGuidanceService(
-            IFileStorageService fileStorageService,
-            IDataGuidanceSubjectService dataGuidanceSubjectService)
+            IDataGuidanceSubjectService dataGuidanceSubjectService,
+            Interfaces.IPublicationService publicationService,
+            Interfaces.IReleaseService releaseService)
         {
-            _fileStorageService = fileStorageService;
             _dataGuidanceSubjectService = dataGuidanceSubjectService;
+            _publicationService = publicationService;
+            _releaseService = releaseService;
         }
 
-        public async Task<Either<ActionResult, DataGuidanceViewModel>> Get(string publicationPath, string releasePath)
+        public async Task<Either<ActionResult, DataGuidanceViewModel>> Get(string publicationSlug, string? releaseSlug = null)
         {
-            var publicationTask = _fileStorageService.GetDeserialized<CachedPublicationViewModel>(publicationPath);
-            var releaseTask = _fileStorageService.GetDeserialized<CachedReleaseViewModel>(releasePath);
+            var publicationTask = _publicationService.Get(publicationSlug);
```

Comment:
I've changed this in all places that a `Task.WhenAll` is used. There is one instance where I believe we could still use `Task.WhenAll` (`ReleaseService#GetSummary`), but I figure the future risk isn't worth the performance benefit, which I suspect is negligible. So to prevent a future bug I've awaited `_publicationService.Get` and `GetCachedRelease` separately there too.